src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -584,18 +584,18 @@ bool Connection::websocket_closed_handler(bool was_clean, WebSocketError error_c
             break;
         }
         case WebSocketError::websocket_fatal_error: {
+            // Error is fatal if the sync_route has already been verified
+            SessionErrorInfo error_info(
+                {ErrorCodes::SyncConnectFailed, util::format("Failed to connect to sync: %1", msg)},
+                IsFatal{m_server_endpoint.is_verified});
+            ConnectionTerminationReason reason = ConnectionTerminationReason::http_response_says_fatal_error;
```

Comment:
is this the correct reason if the errors is not fatal (i.e, is_verified = false)?

Reply:
Yes - if the location hasn't been updated since the `App` was created, then we want this to not be a fatal error and perform a location update to try to request the correct sync route from the base URL address. Until the location info has been requested successfully, this error will continue to be treated as a nonfatal error and will retry to update the location info using the normal backoff delay.

Until the location info is verified, we're not sure if we got the failure because the URL was generated incorrectly or if the server is actually down/misconfigured.

Reply:
I updated the comment to make it more clear

Reply:
I see. My comment was mainly about `http_response_says_fatal_error` being used even if the error is non-fatal.